packages/shell-api/src/bulk.spec.ts

```diff
@@ -114,29 +120,6 @@ describe('Bulk API', () => {
                 nInsertOps: 0, nUpdateOps: 0, nRemoveOps: 0, nBatches: 4
               });
             });
-            it('returns unknown if batches cannot be counted', () => {
-              const bulk2 = new Bulk({} as any, { insert: () => {} } as any, t === 'ordered').insert({}).insert({});
-              expect(bulk2.tojson()).to.deep.equal({
-                nInsertOps: 2, nUpdateOps: 0, nRemoveOps: 0, nBatches: 'unknown'
-              });
-            });
-            it('counts current batches', () => {
```

Comment:
This test should be alright to keep, right?

Reply:
Well … it becomes fairly artificial at that point, something like this:

```js
it('counts current batches', () => {
  const batches: any = [{}];
  const bulk2 = new Bulk({} as any, {
    insert: () => {
      if (t === 'unordered') {
        batches.push({});
      }
    },
    batches
  } as any,
  t === 'ordered'
  ).insert({}).insert({});
  expect(bulk2.tojson()).to.deep.equal({
    nInsertOps: 2, nUpdateOps: 0, nRemoveOps: 0, nBatches: t === 'ordered' ? 1 : 3
  });
});
```

and at that point, it doesn’t seem like we’re actually testing the `Bulk` class anymore?

Reply:
Yeah good point :)